app-modules/assistant/src/Filament/Pages/AIAssistant.php

```diff
@@ -39,11 +42,29 @@ class AIAssistant extends Page
 
     public string $error = '';
 
+    public ConsentAgreement $consentAgreement;
+
+    public bool $consentedToTerms = false;
+
+    public bool $loading = true;
+
+    public static function shouldRegisterNavigation(): bool
+    {
+        /** @var User $user */
+        $user = auth()->user();
+
+        return $user->can('assistant.access');
+    }
+
     public function mount(): void
     {
         /** @var User $user */
         $user = auth()->user();
 
+        $this->authorize('assistant.access');
+
+        $this->consentAgreement = ConsentAgreement::where('type', ConsentAgreementType::AZURE_OPEN_AI)->first();
```

Comment:
```suggestion
        $this->consentAgreement = ConsentAgreement::query()
            ->where('type', ConsentAgreementType::AZURE_OPEN_AI)
            ->first();
```

Reply:
Is there a point in adding `query` here? With our IDE helper configuration, `where` should be auto-completable and static analysis should be aware of it.

Reply:
Personally, I think the chaining looks cleaner when the magic static calls are avoided. Your call, but it's also nice to be consistent and always start a query with `::query()`.

Reply:
Hmm, I see your point about the cleanliness of the line. But I always felt like it was unnecessary, especially if you have the IDE helper.

@dgoetzit do you have a preference?